corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -137,6 +137,31 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         return curr;
     }
 
+    function isStyleNPerRowRepeat(styleStr) {
+        const pattern = new RegExp(`\\d+${constants.PER_ROW_REPEAT}(\\s|$)`);
+        const matchingStyles = getMatchingStyles(pattern, styleStr);
+        return matchingStyles.length > 0;
+    }
+
+    function getNPerRowStyleFromRepeatStyle(style) {
+        var parts = style.split('-');
+        var integerPart = parts[0];
+        return integerPart + '-per-row';
+    }
+
+    function isNPerRowRepeatElement(element) {
```

Comment:
This seems like a code smell to me. Can't this be done in a way similar to how we're checking for any other appearance attribute without doing all these specific checks?

Reply:
Do you mean something like:
```
function isNPerRowRepeatElement(element) {
    const styleStr = (style) ? ko.utils.unwrapObservable(element.style.raw) : null;
    if(styleStr) {
        return isStyleNPerRowRepeat(styleStr)
    } else {
        return false
    }
}
```

Reply:
ya something like that, and can probably just pass in the `style` instead of the whole `element` right?

Reply:
Not moving content of `inisStyleNPerRowRepeat` as it is used in `getNPerRowStyleFromRepeatStyle` as well.

Modified in https://github.com/dimagi/commcare-hq/pull/33979/commits/3306c7c5d73ebe0c42d0d40c03213b5fe785945c

Reply:
https://github.com/dimagi/commcare-hq/pull/33979/commits/caa335077b3e082cc0e96b8f226d1fc84a483821